ckan/config/middleware/flask_app.py

```diff
@@ -217,6 +238,8 @@ def ckan_before_request():
     # Update app_globals
     app_globals.app_globals._check_uptodate()
 
+    g.controller, g.action = request.url_rule.endpoint.split('.')[:2]
```

Comment:
This variables are used inside helpers that build navigation links. Without them [this](https://github.com/ckan/ckan/blob/master/ckan/lib/helpers.py#L689) line ends up with error. 
As a quick solution, i use [this code](https://github.com/ckan/ckan/pull/3775/commits/e3cb97851f8d2d6cd64f6d35affcb6a3f57effea), but i'm not sure, whether it will work for index actions of new blueprints..